R/compute_portfolio_returns.R

```diff
@@ -114,13 +120,13 @@ compute_portfolio_returns <- function(
     cli::cli_warn("No 'breakpoint_options_secondary' specified in bivariate sort.")
   }
 
-  required_columns <- c(sorting_variables, "ret_excess")
+  required_columns <- c(sorting_variables, data_options[["date"]], data_options[["id"]], data_options[["ret_excess"]])
   missing_columns <- setdiff(required_columns, colnames(sorting_data))
   if (length(missing_columns) > 0) {
     cli::cli_abort("The 'sorting_data' is missing the following required columns: {paste(missing_columns, collapse = ', ')}.")
   }
 
-  mktcap_lag_missing <- !"mktcap_lag" %in% colnames(sorting_data)
+  mktcap_lag_missing <- !(data_options$mktcap_lag %in% colnames(sorting_data))
```

Comment:
Is there a reason why we mix $. and [["."]]. I do not see a reason that speaks for one or the other, but I think consistency could be helpful.